test/functional/test-xhr.js

```diff
@@ -268,15 +268,13 @@ describe('XHR', function() {
               });
         });
 
-        it('should not return a resolved indented promise', () => {
+        it('should not resolve after rejecting promise', () => {
           mockXhr.status = 500;
           mockXhr.responseText = '{"a": "hello"}';
           mockXhr.headers['Content-Type'] = 'application/json';
           mockXhr.getResponseHeader = () => 'application/json';
-          const promise = assertSuccess(
-              createResponseInstance('{"a": 2}', mockXhr));
-          promise.should.be.rejected;
```

Comment:
Isn't the issue that you're not using `.should.eventually.be.rejected`? Or, you could return the promise with a flip-flop:

```js
return assertSuccess(createResponseInstance('{"a": 2}', mockXhr)).then(() => {
  throw new Error('should never reach this'); // This makes a resolved promise blow up the tests
}, (err) => {
  return; // this makes the rejected promise pass the tests
});
```

Importantly, this won't hide a never resolving promise.